nixos/modules/services/networking/firewall.nix

```diff
@@ -135,18 +159,102 @@ let
       ip46tables -t raw -A PREROUTING -j nixos-fw-rpfilter
     ''}
 
+    # The "nixos-fw-[input|output|forward]" chains do the actual work.
+    ip46tables -N nixos-fw-input
+    ip46tables -N nixos-fw-output
+    ip46tables -N nixos-fw-forward
+
+    ip46tables -N __invalid_chain__
+
+    # Accept packets from established or related connections.
+    ip46tables -A nixos-fw-input -m conntrack --ctstate ESTABLISHED,RELATED -j nixos-fw-accept
+    ip46tables -A nixos-fw-forward -m conntrack --ctstate ESTABLISHED,RELATED -j nixos-fw-accept
+    ip46tables -A nixos-fw-output -m conntrack --ctstate ESTABLISHED,RELATED -j nixos-fw-accept
```

Comment:
I wouldn't touch forward by default (probably set the policy to `DROP` - but that could break some setups).

I assume new connections are handled below, if not we need the connection state `NEW` as well.